sympy/utilities/misc.py

```diff
@@ -373,7 +373,10 @@ def translate(s, a, b=None, c=None):
 
     mr = {}
     if a is None:
-        assert c is None
+        if c is not None:
+            raise TypeError('Called translate with 4 arguments and a second '
```

Comment:
Better use a `ValueError` here. With error message
`"c should be None when a=None is passed, instead got %s" % c`